### DISCUSSION TITLE
Importing the .bin directly?

### DISCUSSION THREAD

**Discussion by eLiidyr** [isaac-mason/recast-navigation-js#430] [Q&A]
Are you able to import the `.bin` file directly from exported files in recast? I am having trouble getting it to find any kind of `refPoly` or `path` from any file I know works using C library directly. I am also unsure if when you import do you have to run a generator on the import, or can you just query the import directly. I tried to check the examples pages, but I couldn't find any code for the examples

I apologize in advance if I am doing this incorrect.
* Edit - I actually am not in any need of any rendering features, just the calculations of pathing only.

```js
const data: Uint8Array = await readFile(path.join(this.meshes_path, zone + '.bin'));
const { navMesh } = importNavMesh(data);
}
```

I eventually realized it works fine. I just had to invert some of the coordinates so it actually was not finding paths. I was under the impression the `.bin` files might be different data or interpreted differently. 👍

**Comment by isaac-mason** [comment]
Hey, apologies for the late response. From your edit it sounds like you've resolved your issue?

Is this export you're using created with `recast-navigation-js`, or with recast directly e.g. RecastDemo?

`recast-navigation-js` should support both but admittedly the repo should add a test for this to make sure it doesn't break.

Let me know if you're still running into issues!

**Comment by eLiidyr** [comment]
Yeah it was used with RecastDemo directly. I just didn't see any examples specifically using files exported from RecastDemo. I forgot that the coordinates for the meshes are inverted in this particular game. Once I did that it did seem to return the correct data. Thank you for the response though! 😃
